### ISSUE TITLE
apt module reports false changes when locale is not english

### ISSUE THREAD

**Issue by lruppert** [ansible/ansible-modules-core#3188]
##### Issue Type:
- Bug Report
##### Plugin Name:

apt
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

No changes
##### Environment:

Debian Jessie 8.3 with an Ubuntu 14.04 target
##### Summary:

When running in a locale that is not English, the apt module reports false changes for an upgrade. Switching the locale to en_US.UTF-8 causes the problem to go away.
##### Steps To Reproduce:

Ensure that French language is on destination system
set LANG variable to fr_FR.UTF-8
Run playbook

```
apt: upgrade=full
```
##### Expected Results:

No changes reported
##### Actual Results:

```
changed: [mail.local.com] => {"changed": true, "invocation": {"module_args": {"cache_valid_time": null, "deb": null, "default_release": null, "dpkg_options": "force-confdef,force-confold", "force": false, "install_recommends": null, "package": null, "purge": false, "state": "present", "update_cache": false, "upgrade": "full"}, "module_name": "apt"}, "msg": "Lecture des listes de paquets...\nConstruction de l'arbre des dépendances...\nLecture des informations d'état...\nLecture de l'information d'état étendu...\nInitialisation de l'état des paquets...\nAucun paquet ne va être installé, mis à jour ou enlevé.\n0 paquets mis à jour, 0 nouvellement installés, 0 à enlever et 0 non mis à jour.\nIl est nécessaire de télécharger 0 o d'archives. Après dépaquetage, 0 o seront utilisés.\nLecture des listes de paquets...\nConstruction de l'arbre des dépendances...\nLecture des informations d'état...\nLecture de l'information d'état étendu...\nInitialisation de l'état des paquets...\n", "stderr": "", "stdout": "Lecture des listes de paquets...\nConstruction de l'arbre des dépendances...\nLecture des informations d'état...\nLecture de l'information d'état étendu...\nInitialisation de l'état des paquets...\nAucun paquet ne va être installé, mis à jour ou enlevé.\n0 paquets mis à jour, 0 nouvellement installés, 0 à enlever et 0 non mis à jour.\nIl est nécessaire de télécharger 0 o d'archives. Après dépaquetage, 0 o seront utilisés.\nLecture des listes de paquets...\nConstruction de l'arbre des dépendances...\nLecture des informations d'état...\nLecture de l'information d'état étendu...\nInitialisation de l'état des paquets...\n", "stdout_lines": ["Lecture des listes de paquets...", "Construction de l'arbre des dépendances...", "Lecture des informations d'état...", "Lecture de l'information d'état étendu...", "Initialisation de l'état des paquets...", "Aucun paquet ne va être installé, mis à jour ou enlevé.", "0 paquets mis à jour, 0 nouvellement installés, 0 à enlever et 0 non mis à jour.", "Il est nécessaire de télécharger 0 o d'archives. Après dépaquetage, 0 o seront utilisés.", "Lecture des listes de paquets...", "Construction de l'arbre des dépendances...", "Lecture des informations d'état...", "Lecture de l'information d'état étendu...", "Initialisation de l'état des paquets..."]}

```

Looking through the source cod for the module, it appears to match on english regex, which would explain the behavior.

**Comment by robinro** [comment]
Yum has this implemented by setting the `LANG` env: https://github.com/ansible/ansible-modules-core/blob/63fda10853123abef3ef110a062fbfd3769fe64d/packaging/os/yum.py#L261

I'm surprised this didn't come up before for apt.

**Comment by mscherer** [comment]
Subversion and git module too. It wasn't see earlier because only upgrade use apt, installation seems to use a python apt module who likely handle that.

**Comment by resmo** [comment]
fixed by 1df8ebf0c32676d531eb5809a66a941c27ad73e3 in devel and 8d6a3b166c87f1182a60f3c1a2f775d05a9666d9 in stable-2.0
